corehq/messaging/scheduling/view_helpers.py

```diff
@@ -159,106 +185,161 @@ def upload(self, workbook):
 
         return self.msgs
 
-    def update_rule(self, rule, row):
+    def update_rule(self, rule, rows):
+        if len(set([r['name'] for r in rows])) != 1:
+            raise RuleUpdateError(_("Rule name must be the same in all rows"))
+
+        row = rows[0]
         dirty = False
         if 'name' in row and rule.name != row['name']:
             dirty = True
             rule.name = row['name']
+
         return dirty
 
-    def update_rule_message(self, rule, message):
+    def update_rule_messages(self, rule, messages):
         schedule = rule.get_messaging_rule_schedule()
         send_frequency = ScheduleForm.get_send_frequency_by_ui_type(schedule.ui_type)
         {
             ScheduleForm.SEND_IMMEDIATELY: self.save_immediate_schedule,
             ScheduleForm.SEND_DAILY: self.save_daily_schedule,
             ScheduleForm.SEND_WEEKLY: self.save_weekly_schedule,
             ScheduleForm.SEND_MONTHLY: self.save_monthly_schedule,
-        }[send_frequency](schedule, message)
+            ScheduleForm.SEND_CUSTOM_IMMEDIATE: self.save_custom_immediate_schedule,
+            ScheduleForm.SEND_CUSTOM_DAILY: self.save_custom_daily_schedule,
+        }[send_frequency](schedule, messages)
 
-    def save_immediate_schedule(self, schedule, message):
+    def save_immediate_schedule(self, schedule, messages):
         AlertSchedule.assert_is(schedule)
-        schedule.set_simple_alert(SMSContent(message=message),
+        assert len(messages) == 1, "Immediate schedule expected 1 message, got " + six.text_type(len(messages))
```

Comment:
nit: string formatting is more concise

```suggestion
        assert len(messages) == 1, "Immediate schedule expected 1 message, got %s" % len(messages)
```